src/utils/auth.ts

```diff
@@ -0,0 +1,40 @@
+import express, {Request, NextFunction} from 'express';
+const jwt = require('jsonwebtoken');
+import { PrismaClient } from '@prisma/client';
+
+const prisma = new PrismaClient();
+
+
+export interface ReqWithUser extends Request {
+    user: {
+        id: number;
+        email: string;
+        dateCreated: Date;
+        username: string;
+        password?: string;
+    }
+}
+
+// TODO: Test out this middleware once login endpoint is done and returns a JSON web token
+const authenticateJWT = async (req: ReqWithUser, next: NextFunction, prisma: PrismaClient) => {
```

Comment:
This isn't correct, the callback for middleware takes req, res, next, and does not take prisma as an argument? Was this working?
```suggestion
const authenticateJWT = async (req: ReqWithUser, res: Response, next: NextFunction) => {
```

Reply:
It was. I forgot that `const prisma = new PrismaClient()` will be declared in every api file.